Makes Selenium2Library compatible with RobotFramework 3.0 and Python 3.4, 2.7 and 2.6

HelioGuilherme66 opened the pull request:
This PR replaces #556 and contains PRs #559, #561, #562 and #563. Please merge them in order and look at commits to see changes grouped by code modification (not always ;).

HelioGuilherme66 committed:
Changes to Travis settings to allow installation dependent on Python versions.
Already has some settings for Python 3.4, but disabled.

HelioGuilherme66 committed:
Transformed print statement into function. Compatible with Python 2.6.

HelioGuilherme66 committed:
Fixes import to be compatible with RobotFramework 3.0.
Changes Travis settings to install RobotFramework 3.0.

HelioGuilherme66 committed:
Separates imports per line, for better code readability.

HelioGuilherme66 committed:
In `except` uses `as` instead of `,` when capturing the exception.

HelioGuilherme66 committed:
With this commit Selenium2Library is compatible with RobotFramework 3.0
 under Python 2.6, 2.7 and 3.4.
Activates Python 3.4 in Travis config.
Changed files with futurize from http://python-future.org/:
   build_dist.py
   demo/demoapp/server.py
   demo/package.py
   doc/buildhtml.py
   setup.py
   src/Selenium2Library/__init__.py
   src/Selenium2Library/keywords/__init__.py
   src/Selenium2Library/keywords/_alert.py
   src/Selenium2Library/keywords/_browsermanagement.py
   src/Selenium2Library/keywords/_cookie.py
   src/Selenium2Library/keywords/_element.py
   src/Selenium2Library/keywords/_formelement.py
   src/Selenium2Library/keywords/_javascript.py
   src/Selenium2Library/keywords/_logging.py
   src/Selenium2Library/keywords/_runonfailure.py
   src/Selenium2Library/keywords/_screenshot.py
   src/Selenium2Library/keywords/_selectelement.py
   src/Selenium2Library/keywords/_tableelement.py
   src/Selenium2Library/keywords/_waiting.py
   src/Selenium2Library/keywords/keywordgroup.py
   src/Selenium2Library/locators/__init__.py
   src/Selenium2Library/locators/customlocator.py
   src/Selenium2Library/locators/elementfinder.py
   src/Selenium2Library/locators/tableelementfinder.py
   src/Selenium2Library/locators/windowmanager.py
   src/Selenium2Library/utils/__init__.py
   src/Selenium2Library/utils/events/__init__.py
   src/Selenium2Library/utils/events/event.py
   src/Selenium2Library/utils/events/scope_event.py
   src/Selenium2Library/utils/librarylistener.py
   src/Selenium2Library/webdrivermonkeypatches.py
   src/ez_setup.py
   test/lib/mockito/__init__.py
   test/lib/mockito/inorder.py
   test/lib/mockito/invocation.py
   test/lib/mockito/matchers.py
   test/lib/mockito/mock_registry.py
   test/lib/mockito/mocking.py
   test/lib/mockito/mockito.py
   test/lib/mockito/spying.py
   test/lib/mockito/verification.py
   test/resources/statuschecker.py
   test/resources/testserver/testserver.py
   test/run_tests.py
   test/unit/keywords/test_browsermanagement.py
   test/unit/utils/test_browsercache.py

just-be-dev commented on doc/generate.py:
```diff
@@ -1,14 +1,17 @@
 except:
     def main():
-        print """Robot Framework 2.7 or later required for generating documentation"""
+        print('"""Robot Framework 2.7 or later required for generating documentation"""')
```
Let's remove all the extraneous quotes here.

HelioGuilherme66 replied:
I left the quotes there because I think that it is intended to print a comment.

just-be-dev commented:
Have you looked through the 3.4 errors? There seems to be a lot of them... Lots of elements couldn't be found.

HelioGuilherme66 committed:
Adds 'future' dependency and information to install it.

HelioGuilherme66 committed:
Removes dependency in setup.py.
Tested in Python 2.6, 2.7 and 3.4.
(thanks to @rickypc)

HelioGuilherme66 committed:
Adapt to correct Python version.

HelioGuilherme66 commented:
In my last commit  (8e95b3f) I changed the way how python is called, by using the version suffix. There are problems with Python3.4 ../http/server.py code. That is why the Travis is failing with Python3.4. You may verify that the demo starts the server correctly (and also other tests I have).
This is the only blocking point for us to have a green Travis build in Python3.4.

HelioGuilherme66 committed:
Fixes testserver with Python3.4. All files are now opened in mode, 'rb'.

pekkaklarck commented:
I'm sorry but this PR is so big and touches so many places that I don't have time to review and comment it in detail. Some general comments:
- In my opinion scripts related to packaging, demo, etc. should not be touched at this point. It seems to me most of them are not used anymore and should just be removed. If some are still needed, they don't urgently need Python 3 support.
- Travis configuration changes should get their own PR. I don't know too much about Travis and canot merge PRs touching it without knowing what change do.
- As I have commented earlier, I would like to see separate PRs fixing individual Python 3 compatibility issues without affecting Python 2 compatibility. Two easy topics to start are:
  - Exceptions: `except Exception, err` -> `except Exception as err`.
  - Imports: `from internal import Stuff` -> `from .internal import Stuff`. This PR could also contain general cleanup to imports to make them more PEP-8 compatible. It shouldn't contain any other code changes.

HelioGuilherme66 committed:
Add GitHub process files.

HelioGuilherme66 committed:
Add GitHub process files.

Update CONTRIBUTING.md

Fixes test run in Fedora.

HelioGuilherme66 committed:
Merge pull request #5 from HelioGuilherme66/gitdocs

Add GitHub process files.

HelioGuilherme66 committed:
Update CONTRIBUTING.md

Fixed Issues URL to point to my fork.

HelioGuilherme66 committed:
Update ISSUE_TEMPLATE.md

Fix typo

HelioGuilherme66 committed:
Upgrades Travis firefox to latest-esr.

HelioGuilherme66 committed:
Fixes test run in Fedora.

HelioGuilherme66 committed:
Keep selenium==2.52.0 in Travis.
Version >=2.53.0 is breaking Unselect From List By Value.

HelioGuilherme66 committed:
Fixes ez_setup.py. Changes version to 1.8.0b2.

HelioGuilherme66 committed:
Merge pull request #10 from HelioGuilherme66/fix_unselect

Fixes `Unselect From List` due to Selenium changes.

HelioGuilherme66 committed:
Removes selenium version dependency, current version is 2.53.2.

HelioGuilherme66 committed:
Merge branch 'master' of https://github.com/robotframework/Selenium2Library

HelioGuilherme66 committed:
Merge branch 'master' of https://github.com/robotframework/Selenium2Library into code_futurized

HelioGuilherme66 committed:
Merge branch 'master' of https://github.com/robotframework/Selenium2Library

HelioGuilherme66 committed:
Merge pull request #7 from HelioGuilherme66/master

Testing Pull Request Template

aaltat commented:
I think Pekka comment is still a valid one. And also the commit history is awful.

HelioGuilherme66 commented:
This PR is the accumulated commits. We should analyze them in order, starting with PR #559, then after that one is merged, we go for #561, and so on. The latest commits in this one, could be discarded as they are essentially merges from master.

HelioGuilherme66 committed:
Merge branch 'master' into code_futurized

	modified:   CONTRIBUTING.md
	modified:   src/Selenium2Library/keywords/_browsermanagement.py

HelioGuilherme66 committed:
Merge branch 'master' of https://github.com/robotframework/Selenium2Library

HelioGuilherme66 committed:
Merge branch 'master' into code_futurized

HelioGuilherme66 committed:
Now calls run_unit_tests.py file instead of the method (this fixes failure in Fedora Core 22).
Get path to python/jython to run unit tests.

ombre42 committed:
Libdoc updates - link to project and fix 404 (#668)

* Fix 404 due to RF migration to Github

* Link to project page

* Format corrections per PR comments

* fix pronoun usage

HelioGuilherme66 committed:
Changes to use Firefox ESR (45.4.0)

While we are not having Selenium 3.0 working, with external `geckodriver`, we use lower versions of stable Firefox.

HelioGuilherme66 committed:
Changes to use Selenium 2.53.6 (and not 3.0)

We are not ready for Selenium 3.0 yet.

Gaurang033 committed:
Added method to find element by class name #673 (#674)

* Added method to find element by class name #673

* Changes to use Firefox ESR (45.4.0)

While we are not having Selenium 3.0 working, with external `geckodriver`, we use lower versions of stable Firefox.

* Changes to use Selenium 2.53.6 (and not 3.0)

We are not ready for Selenium 3.0 yet.

HelioGuilherme66 committed:
Updated Travis config.

HelioGuilherme66 committed:
Merges some latest changes from master.
Sets this version to 1.8.1b1.

HelioGuilherme66 committed:
Attempt to fix run_tests on Windows.

HelioGuilherme66 committed:
Merge branch 'master' of github.com:HelioGuilherme66/robotframework-selenium2library into code_futurized

pekkaklarck commented:
Most of the changes done here is already done by other PRs. This was too big PR with too many changes. to begin with.
